src/hocon_postprocess.erl

```diff
@@ -170,23 +170,50 @@ duration(Other) ->
 do_duration(Str) ->
     case do_duration(Str, 0) of
         skip -> Str;
-        Int -> Int
+        Num -> round(Num)
     end.
 do_duration(Str, Sum) ->
-    {ok, MP} = re:compile("^([0-9]+)(d|D|h|H|m|M|s|S|ms|MS)([0-9]+.+)"),
+    {ok, MP} = re:compile("^([0-9\.]+)(f|F|w|W|d|D|h|H|m|M|s|S|ms|MS)([0-9\.]+.+)"),
     case re_run_first(Str, MP) of
         {match, [Val, Unit, Next]} ->
-            do_duration(Next, Sum + calc_duration(list_to_integer(Val), string:lowercase(Unit)));
+            case get_decimal(Val) of
+                D when is_integer(D) orelse is_float(D) ->
+                    do_duration(Next, Sum + calc_duration(D, string:lowercase(Unit)));
+                _ ->
+                    skip
+            end;
         nomatch ->
-            {ok, LastMP} = re:compile("^([0-9]+)(d|D|h|H|m|M|s|S|ms|MS)$"),
+            {ok, LastMP} = re:compile("^([0-9\.]+)(f|F|w|W|d|D|h|H|m|M|s|S|ms|MS)$"),
             case re_run_first(Str, LastMP) of
                 {match, [Val, Unit]} ->
-                    Sum + calc_duration(list_to_integer(Val), string:lowercase(Unit));
+                    case get_decimal(Val) of
+                        D ->
+                            Sum + calc_duration(D, string:lowercase(Unit))
+                    end;
                 nomatch ->
                     skip
             end
     end.
 
+get_decimal([$. | _]=Num) ->
+    get_decimal(["0" | Num]);
+get_decimal(Num) ->
+    case string:to_integer(Num) of
+        {I, []} ->
+            I;
+        {Int, [$. | MaybeDecimal]} ->
+            case string:to_integer(MaybeDecimal) of
+                {Decimal, []} ->
+                    Int + Decimal / math:pow(10, length(MaybeDecimal));
+                _ ->
+                    {error, badarg}
+            end;
+        _ ->
+            error
```

Comment:
it's better if `get_decimal` returns either `error` or `{error, Reason}`, but not a mix